app/src/main/java/blockeq/com/stellarwallet/activities/MyWalletActivity.kt

```diff
@@ -12,4 +23,40 @@ class MyWalletActivity : BasePopupActivity() {
         return R.layout.content_my_wallet
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        titleText.text = availableBalance
+        camera_image_button.setOnClickListener { initiateScan() }
+
+        next_button.isEnabled = false
+        next_button.setOnClickListener {
+
+            val intent = Intent(this, SendActivity::class.java).apply {
+                putExtra(ADDRESS_DATA, address_text.text)
+            }
+            startActivity(intent)
+
+            this.overridePendingTransition(R.anim.slide_in_up, R.anim.stay)
+
+        }
+    }
+
+    private fun initiateScan() {
+        IntentIntegrator(this).setBeepEnabled(false).setDesiredBarcodeFormats(IntentIntegrator.QR_CODE).initiateScan()
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        val result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
```

Comment:
Scanner works great. However, when the user hits the back button it crashes. We probably just need to handle that case in the ActivityResult and we should be good to go. This GIF will help in reproducing this:

![cameracancel](https://user-images.githubusercontent.com/2676558/43088363-f843a69e-8e6f-11e8-8ad5-7b40a5056b37.gif)